docs/JSON-RPC.md

```diff
@@ -0,0 +1,435 @@
+See `tools/generate_json_rpc_docs.py` for details.
+-->
+
+A JSON-RPC interface is available for both Jamulus client and server to allow programmatic access.
+To use the JSON-RPC interface, run Jamulus with the `--jsonrpcport <port> --jsonrpcsecretfile /file/with/a/secret.txt` options.
+This will start a JSON-RPC server on the specified port on the localhost.
+
+## Wire protocol
+
+The JSON-RPC server is based on the [JSON-RPC 2.0](https://www.jsonrpc.org/specification) protocol, using [streaming newline-delimited JSON over TCP](https://clue.engineering/2018/introducing-reactphp-ndjson) as the transport. There are three main types of messages being exchanged:
+
+- A **request** from the consumer to Jamulus.
+- A **response** from Jamulus to the consumer.
+- A **notification** from Jamulus to the consumer.
+
+## Example
+
+After opening a TCP connection to the JSON-RPC server, the connection must be authenticated:
+
+```json
+{"id":1,"jsonrpc":"2.0","method":"jamulus/apiAuth","params":{"secret": "...the secret from the file in --jsonrpcsecretfile..."}}
+```
+
+Request must be sent as a single line of JSON-encoded data, followed by a newline character. Jamulus will send back a **response** in the same manner:
+
+```json
+{"id":1,"jsonrpc":"2.0","result":"ok"}
+```
+After successful authentication, the following **request** can be sent:
+
+```json
+{"id":2,"jsonrpc":"2.0","method":"jamulus/getMode","params":{}}
+```
+
+The request must be sent as a single line of JSON-encoded data, followed by a newline character. Jamulus will send back a **response** in the same manner:
+
+```json
+{"id":2,"jsonrpc":"2.0","result":{"mode":"client"}}
+```
+
+Jamulus will also send **notifications** to the consumer:
+
+```json
+{"jsonrpc":"2.0","method":"jamulusclient/chatTextReceived","params":{"text":"<font color=\"mediumblue\">(01:23:45 AM) <b>user</b></font> test"}}
+```
+
+## Method reference
+### jamulus/apiAuth
+
+Authenticates the connection which is a requirement for calling further methods.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result | string | "ok" on success |
+
+
+### jamulus/getMode
+
+Returns the current mode, i.e. whether Jamulus is running as a server or client.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result.mode | string | The current mode (server or client). |
+
+
+### jamulus/getVersion
+
+Returns Jamulus version.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result.version | string | The Jamulus version. |
+
+
+### jamulusclient/getChannelInfo
+
+Returns your profile information.
```

Comment:
Your profile sounds strange. Maybe better the clients' profile information.

Reply:
Ok, updated.